sdk/servicebus/azure-messaging-servicebus-track1-perf/src/main/java/com/microsoft/azure/messaging/servicebus/perf/ReceiveAndLockMessageTest.java

```diff
@@ -0,0 +1,98 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Performance test.
+ */
+public class ReceiveAndLockMessageTest extends ServiceTest<ServiceBusStressOptions> {
+    private final ClientLogger logger = new ClientLogger(ReceiveAndLockMessageTest.class);
+    private final ServiceBusStressOptions options;
+
+    /**
+     * Creates test object
+     * @param options to set performance test options.
+     */
+    public ReceiveAndLockMessageTest(ServiceBusStressOptions options) {
+        super(options, ReceiveMode.PEEKLOCK);
+        this.options = options;
+    }
+
+    private Mono<Void> sendMessage() {
+        return Mono.defer(() -> {
+            int total = options.getParallel() * options.getMessagesToSend() * TOTAL_MESSAGE_MULTIPLIER;
+            List<Message> messages = new ArrayList<>();
+            for (int i = 0; i < total; ++i) {
+                Message message = new Message(CONTENTS);
+                message.setMessageId(UUID.randomUUID().toString());
+                messages.add(message);
+            }
+            return Mono.fromFuture(sender.sendBatchAsync(messages));
+        });
+    }
+
+    @Override
+    public Mono<Void> globalSetupAsync() {
+        // Since test does warm up and test many times, we are sending many messages, so we will have them available.
+        return super.globalSetupAsync()
+            .then(sendMessage());
+    }
+
+    @Override
+    public void run() {
+        Collection<IMessage> messages = null;
+        try {
+            messages = receiver.receiveBatch(options.getMessagesToReceive());
+        } catch  (Exception e) {
+            throw logger.logExceptionAsWarning(new RuntimeException(e));
+        }
+
+        for (IMessage message : messages) {
+            try {
+                receiver.complete(message.getLockToken());
+            } catch (InterruptedException | ServiceBusException e) {
+                throw logger.logExceptionAsWarning(new RuntimeException(e));
+            }
+        }
+    }
+
+    @Override
+    public Mono<Void> runAsync() {
+        return Mono.fromFuture(receiver.receiveBatchAsync(options.getMessagesToReceive())
+            .thenAccept(iMessages -> {
+                try {
+                    for (IMessage message : iMessages) {
+                        receiver.completeAsync(message.getLockToken()).get();
+                    }
+                } catch (InterruptedException | ExecutionException e) {
+                    throw logger.logExceptionAsWarning(new RuntimeException(e));
+                }
+            }));
+    }
+
+    /**
+     * Runs the cleanup logic after the performance test finishes.
+     * @return An empty {@link Mono}
+     */
+    @Override
+    public Mono<Void> globalCleanupAsync() {
+        return Mono.fromFuture(sender.closeAsync().thenCombine(receiver.closeAsync(), (aVoid, aVoid2) -> true))
```

Comment:
If the future fails, the global cleanup will not run... Why not `Mono.when()` same with other instances.